sdk/communication/communication-phone-numbers/src/sipRoutingClient.ts

```diff
@@ -0,0 +1,298 @@
+
+    const internalPipelineOptions: InternalPipelineOptions = {
+      ...options,
+      ...{
+        loggingOptions: {
+          logger: logger.info,
+        },
+      },
+    };
+
+    this.client = new SipRoutingGeneratedClient(url, {
+      endpoint: url,
+      ...internalPipelineOptions,
+    });
+    const authPolicy = createCommunicationAuthPolicy(credential);
+    this.client.pipeline.addPolicy(authPolicy);
+  }
+
+  /**
+   * Gets the SIP trunks.
+   * @param options - The options parameters.
+   */
+  public async getTrunks(options: OperationOptions = {}): Promise<SipTrunk[]> {
+    const { span, updatedOptions } = createSpan("SipRoutingClient-GetTrunks", options);
+
+    try {
+      return this.client
+        .getSipConfiguration(updatedOptions)
+        .then((config) => mapTrunks(config.trunks));
+    } catch (e: any) {
+      span.setStatus({
+        code: SpanStatusCode.ERROR,
+        message: e.message,
+      });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Gets the SIP trunk.
+   * @param fqdn - The trunk's FQDN.
+   * @param options - The options parameters.
+   */
+  public async getTrunk(fqdn: string, options: OperationOptions = {}): Promise<SipTrunk | null> {
+    const { span, updatedOptions } = createSpan("SipRoutingClient-GetTrunk", options);
+
+    try {
+      return this.getTrunks(updatedOptions).then(
+        (trunks) => trunks.find((value: SipTrunk) => value.fqdn === fqdn) || null
+      );
+    } catch (e: any) {
+      span.setStatus({
+        code: SpanStatusCode.ERROR,
+        message: e.message,
+      });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Gets the SIP trunk routes.
+   * @param options - The options parameters.
+   */
+  public async getRoutes(options: OperationOptions = {}): Promise<SipTrunkRoute[]> {
+    const { span, updatedOptions } = createSpan("SipRoutingClient-GetRoutes", options);
+
+    try {
+      return this.client.getSipConfiguration(updatedOptions).then((config) => config.routes || []);
+    } catch (e: any) {
+      span.setStatus({
+        code: SpanStatusCode.ERROR,
+        message: e.message,
+      });
+      throw e;
+    } finally {
+      span.end();
+    }
+  }
+
+  /**
+   * Sets the SIP trunks.
+   * @param trunks - The SIP trunks to be set.
+   * @param options - The options parameters.
+   */
+  public async setTrunks(trunks: SipTrunk[], options: OperationOptions = {}): Promise<SipTrunk[]> {
+    const { span, updatedOptions } = createSpan("SipRoutingClient-SetTrunks", options);
+
+    try {
+      const patch: SipConfigurationPatch = { trunks: mapTrunksToRestModel(trunks) };
+      const setFqdns = trunks.map((trunk) => trunk.fqdn);
+      const storedFqdns = await this.client
+        .getSipConfiguration(updatedOptions)
+        .then((config) => mapTrunks(config.trunks))
+        .then((value) => value.map((trunk) => trunk.fqdn));
```

Comment:
nit: I found the readability is slightly better if we don't mix await and callbacks
```suggestion
      const config = await this.client
        .getSipConfiguration(updatedOptions);
      const storedFqdns = mapTrunks(config.trunks).map((trunk) => trunk.fqdn);
```

Reply:
Updated at multiple places in sipRoutingClient